### ISSUE TITLE
Official build break: gc.cpp(1827): warning C4800: 'BOOL': forcing value to bool 'true' or 'false' (performance warning)

### ISSUE THREAD

**Issue by RussKeldorph** [dotnet/coreclr#16908]
(https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_build?_a=summary&buildId=1470600)

Suspect related to 8edb3ef2529cc0beb0c034482fdb32fa4c95c606

```
2018-03-13T11:07:15.9868107Z e:\a\_work\371\s\src\gc\gc.cpp(1827): error C2220: warning treated as error - no 'object' file generated (compiling source file E:\A\_work\371\s\src\gc\gcwks.cpp) [E:\A\_work\371\s\bin\obj\Windows_NT.x64.Release\src\gc\clrgc.vcxproj]
2018-03-13T11:07:15.9876005Z e:\a\_work\371\s\src\gc\gc.cpp(1827): warning C4800: 'BOOL': forcing value to bool 'true' or 'false' (performance warning) (compiling source file E:\A\_work\371\s\src\gc\gcwks.cpp) [E:\A\_work\371\s\bin\obj\Windows_NT.x64.Release\src\gc\clrgc.vcxproj]
2018-03-13T11:07:16.0409127Z   gceewks.cpp
2018-03-13T11:07:16.1427293Z e:\a\_work\371\s\src\gc\gc.cpp(1827): error C2220: warning treated as error - no 'object' file generated (compiling source file E:\A\_work\371\s\src\gc\gcsvr.cpp) [E:\A\_work\371\s\bin\obj\Windows_NT.x64.Release\src\gc\clrgc.vcxproj]
2018-03-13T11:07:16.1435581Z e:\a\_work\371\s\src\gc\gc.cpp(1827): warning C4800: 'BOOL': forcing value to bool 'true' or 'false' (performance warning) (compiling source file E:\A\_work\371\s\src\gc\gcsvr.cpp) [E:\A\_work\371\s\bin\obj\Windows_NT.x64.Release\src\gc\clrgc.vcxproj]
[More errors as well]
```

I suspect this would be resolved when the official builds switch to VS2017, but we're not quite there yet.

@davmason @dotnet/dotnet-diag

**Comment by davmason** [comment]
Fixed in #16914

**Comment by adiaaida** [comment]
Thanks for this fix. This was also failing on the perf runs since the perf machines only have VS2015 on them.
